browser/ui/views/omnibox/brave_search_conversion_promotion_view.cc

```diff
@@ -0,0 +1,589 @@
+  SkColor desc_color = gfx::kPlaceholderColor;
+  SkColor border_color = gfx::kPlaceholderColor;
+  if (auto* tp = GetThemeProvider()) {
+    desc_color = tp->GetColor(
+        BraveThemeProperties::COLOR_SEARCH_CONVERSION_BANNER_TYPE_DESC_TEXT);
+    border_color =
+        tp->GetColor(BraveThemeProperties::
+                         COLOR_SEARCH_CONVERSION_BANNER_TYPE_BACKGROUND_BORDER);
+  }
+  banner_type_container_->SetBorder(
+      views::CreateRoundedRectBorder(1, kBannerTypeRadius, border_color));
+  banner_type_description_->SetEnabledColor(desc_color);
+
+  SetBackground(views::CreateSolidBackground(
+      GetOmniboxColor(GetThemeProvider(), OmniboxPart::RESULTS_BACKGROUND)));
+  banner_type_container_->SetBackground(
+      std::make_unique<HorizontalGradientBackground>());
+}
+
+void BraveSearchConversionPromotionView::ConfigureForButtonType() {
+  if (button_type_container_)
+    return;
+
+  button_type_container_ = AddChildView(std::make_unique<views::View>());
+  button_type_container_->SetProperty(
+      views::kFlexBehaviorKey,
+      views::FlexSpecification(views::MinimumFlexSizeRule::kScaleToZero,
+                               views::MaximumFlexSizeRule::kUnbounded));
+
+  button_type_container_
+      ->SetLayoutManager(std::make_unique<views::FlexLayout>())
+      ->SetOrientation(views::LayoutOrientation::kHorizontal);
+
+  auto* selection_container =
+      button_type_container_->AddChildView(std::make_unique<views::View>());
+  selection_container->SetPreferredSize(gfx::Size(3, height()));
+  selection_container->SetLayoutManager(std::make_unique<views::FillLayout>());
+  button_type_selection_indicator_ = selection_container->AddChildView(
+      std::make_unique<BraveOmniboxResultSelectionIndicator>(this));
+
+  auto& bundle = ui::ResourceBundle::GetSharedInstance();
+  auto* icon_view =
+      button_type_container_->AddChildView(std::make_unique<views::ImageView>(
+          ui::ImageModel::FromImageSkia(*bundle.GetImageSkiaNamed(
+              IDR_BRAVE_SEARCH_LOGO_IN_SEARCH_PROMOTION))));
+  icon_view->SetProperty(views::kMarginsKey, gfx::Insets::TLBR(14, 12, 36, 8));
+
+  auto* contents_container =
+      button_type_container_->AddChildView(std::make_unique<views::View>());
+  contents_container->SetProperty(views::kMarginsKey,
+                                  gfx::Insets::TLBR(12, 12, 12, 0));
+  contents_container->SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kVertical, gfx::Insets(), 2));
+
+  auto* input_container =
+      contents_container->AddChildView(std::make_unique<views::View>());
+  input_container->SetLayoutManager(std::make_unique<views::BoxLayout>(
+      views::BoxLayout::Orientation::kHorizontal, gfx::Insets(), 8));
+
+  // Use 14px font size for input text.
+  views::Label::CustomFont emphasized_font = {
+      GetFont(14, gfx::Font::Weight::SEMIBOLD)};
+  button_type_contents_input_ = input_container->AddChildView(
```

Comment:
In the screenshot this looks like it's the same or very similar text format as the regular search result row. Perhaps you could explain here, or leave a comment in the code as to why we're not using `OmniboxMatchCellView` or even at the least `OmniboxTextView` for the "button" version? I'm not very familiar with the omnibox views, so sorry if it's obvious why we can't use them. But I would have hoped we could do less custom views, and extend the existing ones. I'm sure there's a good reason for you to do it like this, so maybe leave a code comment, perhaps at the beginning of the `ConfigureForButtonType` method.

Reply:
Yeah, it seems very similar but has slightly different for our purpose such as layout (you mentioned - buttons posiiton), bg, text colors or font. So, I decided to implement whole UI. In this case, I could set our view to whole area easily by only controlling upstream's view and our view's visibilty w/o touching upstream views.